src/helpers/token.ts

```diff
@@ -0,0 +1,12 @@
+import ms from 'ms';
+import { sign } from 'jsonwebtoken';
+import { JWT_SECRET, JWT_EXPIRATION_PERIOD } from '../env';
+
+export const format = (token: string) => `${token.slice(0, 3)}...${token.slice(-3)}`;
+
+export const generate = (sessionId: string, expireDate?: Date) => {
+  const expiresIn = expireDate
+    ? ms(expireDate.getTime() - Date.now())
+    : JWT_EXPIRATION_PERIOD;
```

Comment:
To be more precise, I would do the opposite: 
- I would accept duration in originally supplied format here
- Return jwt token
- Provide another function which parses jwt token and returns `exp` (encoded into it) as a js Date
- Write Date parsed from the token into the database

Reply:
Replaced with number of seconds. 

The last 2 bullet points dont make sense to me, why would i write data about expiry after parsing the token if i already have the data about expiry before encoding the token? would it not be easier to map the seconds into expiry date directly without passing it through JWT encode-decode?

Reply:
> why would i write data about expiry after parsing the token if i already have the data about expiry before encoding the token?

Well, since you don't control what's written into the JWT, because you don't how they process supplied period and turn it into unix timestamp, how they handle timezones, etc. It can be, that you provide one thing, but something else written to the token (eg you provide no expiration date, but something else is written there), so to be factual you should decode it as suggested